tip/tip.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+
+import re
+from sys import argv, stdout, exit as sys_exit
+from os import environ, getenv
+from glob import glob
+from pathlib import Path
+from github import Github
+from github import GithubException
+
+print("· Get list of artifacts to be uploaded")
+
+args = []
+files = []
+
+if "INPUT_FILES" in environ:
+    args = environ["INPUT_FILES"].split()
+
+if len(argv) > 1:
+    args = args + argv[1:]
+
+if len(args) == 1 and args[0] == "none":
+    files = []
+    print("! Skipping 'files' because it's set to 'none")
+else:
+    if len(args) == 0:
+        stdout.flush()
+        raise (Exception("Glob patterns need to be provided as positional arguments or through envvar 'INPUT_FILES'!"))
+
+    for item in args:
+        items = [fname for fname in glob(item, recursive=True) if not Path(fname).is_dir()]
+        print(f"  glob({item!s}):")
+        for fname in items:
+            if Path(fname).stat().st_size == 0:
+                print(f"  - ! Skipping empty file {fname!s}")
+                continue
+            print(f"  - {fname!s}")
+            files += [fname]
+
+    if len(files) < 1:
+        stdout.flush()
+        raise (Exception("Empty list of files to upload/update!"))
+
+print("· Get GitHub API handler (authenticate)")
+
+if "GITHUB_TOKEN" in environ:
+    gh = Github(environ["GITHUB_TOKEN"])
+elif "INPUT_TOKEN" in environ:
+    gh = Github(environ["INPUT_TOKEN"])
+else:
+    if "GITHUB_USER" not in environ or "GITHUB_PASS" not in environ:
+        stdout.flush()
+        raise (
+            Exception(
+                "Need credentials to authenticate! Please, provide 'GITHUB_TOKEN', 'INPUT_TOKEN', or 'GITHUB_USER' and 'GITHUB_PASS'"
+            )
+        )
+    gh = Github(environ["GITHUB_USER"], environ["GITHUB_PASS"])
+
+print("· Get Repository handler")
+
+if "GITHUB_REPOSITORY" not in environ:
+    stdout.flush()
+    raise (Exception("Repository name not defined! Please set 'GITHUB_REPOSITORY"))
+
+gh_repo = gh.get_repo(environ["GITHUB_REPOSITORY"])
+
+print("· Get Release handler")
+
+tag = getenv("INPUT_TAG", "tip")
+
+env_tag = None
+gh_ref = environ["GITHUB_REF"]
+is_prerelease = True
+is_draft = False
+
+if gh_ref[0:10] == "refs/tags/":
+    env_tag = gh_ref[10:]
+    if env_tag != tag:
+        rexp = r"^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
```

Comment:
Should this be handled through pyTooling.Version.SemVersion?

Reply:
Might be added to pyTooling.Version.SemVersion, because it can be useful in that package as well. However, I don't think it's worth making Tip dependent on pyTooling.Version just for this line. This is in fact copied from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string as-is. 

A possible enhancement to Tip is to build a container image in a separated workflow. Then, Tip itself would not need to install the dependencies at runtime; it would consume a ready-to-use ghcr.io/pyTooling/Tip container. In that context, having additional dependencies would be less harmful (not at all). Moreover, if we did that, we might add a proper CLI to Tip, through pyAttributes.

Reply:
Extracted into an issue: https://github.com/pyTooling/pyTooling/issues/12